smithy-codegen-core/src/main/java/software/amazon/smithy/codegen/core/ShapesGenerationOrdering.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.codegen.core;
+
+/**
+ * Shapes ordering for code generation.
+ *
+ * <p>CodegenDirector order the shapes appropriately before feeding them to the code generators. See {@link
+ * software.amazon.smithy.codegen.core.directed.CodegenDirector#setShapesGenerationOrdering(ShapesGenerationOrdering)}
+ */
+public enum ShapesGenerationOrdering {
```

Comment:
naming is always hard. I'm thinking about, ordering v/s order. different people may have different opinions on this. I say consider `ShapesGenerationOrder`